specs/daito/Multi_cloud_yig_integration.md

```diff
@@ -0,0 +1,325 @@
+
+## Motivation
+
+Currently OpenSDS multi-cloud comes with a set of feature and storage backends.
+
+### Here is the list of features available in OpenSDS:
+Multi-cloud Management across public and private clouds (Data Migration, Lifecycle Management)
+Cloud Vendor Agnostic
+Policy based data mobility
+S3 based REST API
+Containerized deployment
+Metadata Management and monitoring
+Easy integration of backend storage adapters (on-prem or Cloud)
+
+### List of Object Storage backends supported:
+    * AWS
+    * Azure
+    * GCP
+    * Huawei OBS
+    * IBM COS
+    * Ceph S3
+
+### New features to be supported:
+    * Encryption of Data at rest
+    * Bucket Location
+    * Versioning
+    * CORS
+    * ACL
+    * Bucket Policy
+    * Configurable buffered copy object
+    * HEAD Bucket
+    * HEAD Object
+
+### New Object Storage Backend support: YIG
+
+#### Some of the key features of YIG:
+YIG can avoid data movement and I/O drop down caused by scaling up of Ceph cluster 
+With scale-out Ceph-cluster and ceph-libradosstripper APIs, YIG solves some of the key performance issues of Ceph
+These features make YIG Ceph Pool an important on-premise Object Storage backend.
+
+### Why YIG:
+YIG brings in some of the features which can complement and enhance OpenSDS multi-cloud management platform. This also brings in opportunity for OpenSDS to be THE Platform for all multi-cloud management. YIG has some features which OpenSDS multi-cloud doesn't have. So YIG can help adding those features into OpenSDS multi-cloud.
+
+ 
+How YIG can help in enhancing multi-cloud capabilities of Gelato:
+YIG already supports following features for the YIG Ceph pool:
+
+    * Bucket and Object ACL
+    * Bucket policy
+    * Copying Objects
+    * Server-side encryption
+    * Versioning
+    * Post object
+
+### Goals
+This document will propose the feature enhancement to OpenSDS multi-cloud (Gelato) and integrating YIG to achieve this. The new set of features are:
+
+    * Encryption of Data at rest
+    * Bucket Location
+    * Versioning
+    * CORS
+    * ACL
+    * Bucket Policy
+    * Configurable buffered copy object
+    * HEAD Bucket
+    * HEAD Object
+
+### Non-Goals
+
+Features not supported by OpenSDS multi-cloud (Gelato) or YIG are not included.
+
+## Design Details
+    
+### Enhance multi-cloud (Gelato) and integrate with YIG
+
+
+#### For this approach here is the high-level architecture:
+
+![gelato-multicloud-yig](gelato-multicloud-yig.png?raw=true "Gelato Multicloud with YIG")
+
+
```

Comment:
We need some detailed description on how migration works with YIG integration, i.e., what is the role of each component when moving objects from source bucket to destination bucket.